packages/jellyfish-api-core/__tests__/category/loan/listAuctionHistory.test.ts

```diff
@@ -0,0 +1,592 @@
+        expect(auctionhistory1).toStrictEqual(auctionhistory2)
+        expect(auctionhistory1.length).toStrictEqual(2)
+        {
+          const result = auctionhistory1.find(h => h.vaultId === vaultId3)
+          expect(result).toStrictEqual(
+            {
+              winner: bobColAddr,
+              blockHeight: expect.any(Number),
+              blockHash: expect.any(String),
+              blockTime: expect.any(Number),
+              vaultId: vaultId3,
+              batchIndex: 0,
+              auctionBid: '23625.00000000@MSFT',
+              auctionWon: ['6666.66660000@DFI', '0.33333333@BTC']
+            }
+          )
+        }
+        {
+          const result = auctionhistory1.find(h => h.vaultId === vaultId4)
+          expect(result).toStrictEqual(
+            {
+              winner: bobColAddr,
+              blockHeight: expect.any(Number),
+              blockHash: expect.any(String),
+              blockTime: expect.any(Number),
+              vaultId: vaultId4,
+              batchIndex: 1,
+              auctionBid: '31500.00000000@FB',
+              auctionWon: ['6666.66640000@DFI', '0.33333332@BTC']
+            }
+          )
+        }
+      }
+    })
+
+    it('should listAuctionHistory with owner = all', async () => {
+      const data = await alice.rpc.loan.listAuctionHistory('all')
+      expect(data.length).toStrictEqual(4)
+
+      const data1 = data.find(d => d.vaultId === vaultId1)
+      expect(data1).toBeDefined()
+
+      const data2 = data.find(d => d.vaultId === vaultId2)
+      expect(data2).toBeDefined()
+
+      const data3 = data.find(d => d.vaultId === vaultId3)
+      expect(data3).toBeDefined()
+
+      const data4 = data.find(d => d.vaultId === vaultId4)
+      expect(data4).toBeDefined()
+    })
+
+    it('should listAuctionHistory with owner = address', async () => {
+      {
+        const data = await alice.rpc.loan.listAuctionHistory(aliceColAddr)
+        const data1 = data.find(d => d.vaultId === vaultId1)
+        expect(data1).toBeDefined()
+
+        const data2 = data.find(d => d.vaultId === vaultId2)
+        expect(data2).toBeDefined()
+
+        const data3 = data.find(d => d.vaultId === vaultId3)
+        expect(data3).toBeUndefined()
+
+        const data4 = data.find(d => d.vaultId === vaultId4)
+        expect(data4).toBeUndefined()
+      }
+
+      {
+        const data = await alice.rpc.loan.listAuctionHistory(bobColAddr)
+        const data1 = data.find(d => d.vaultId === vaultId1)
+        expect(data1).toBeUndefined()
+
+        const data2 = data.find(d => d.vaultId === vaultId2)
+        expect(data2).toBeUndefined()
+
+        const data3 = data.find(d => d.vaultId === vaultId3)
+        expect(data3).toBeDefined()
+
+        const data4 = data.find(d => d.vaultId === vaultId4)
+        expect(data4).toBeDefined()
+      }
+    })
+  })
+
+  describe('listAuctionHistory with pagination', () => { // For pagination, always filter by 'all'.
```

Comment:
`For pagination, always filter by 'all'.` that is actually not true. you can paginate the results which are filtered by `mine ` or `owner`. and it's real world use case that we filter by `mine ` or `owner` and get the results paginated.
Can add a test for that too.
note that `owner` is filtration and all fields in `ListAuctionHistoryPagination` are for pagination.
Here also database index has `height` as first. So it's crucial that we pass the intended `height` that the pagination should start.

from cpp side
```
inline void SerializationOp(Stream& s, Operation ser_action) {

        if (ser_action.ForRead()) {
            READWRITE(WrapBigEndian(blockHeight));
            blockHeight = ~blockHeight;
        } else {
            uint32_t blockHeight_ = ~blockHeight;
            READWRITE(WrapBigEndian(blockHeight_));
        }

        READWRITE(owner);
        READWRITE(vaultId);
        READWRITE(index);
    }
```

Reply:
Added the test units for 
- 'mine' and address.
- height + vaultId
- height + index

Reply:
Added some minor refactor to pagination tests.